tests/test_chord.py

```
@@ -639,8 +639,10 @@ def test_validate():
     with pytest.raises(ValueError):
         mir_eval.chord.validate([], ["C"])
 
-# @maximoskp
-# make testing for all chord types
-# quals = mir_eval.chord.QUALITIES
-# for q in quals.keys():
-#     c = mir_eval.chord.encode('C' + (len(q) > 0)*':' + q, reduce_extended_chords=True)
\ No newline at end of file
+def test_chord_qualities():
+    quals = mir_eval.chord.QUALITIES
+    for q in quals.keys():
+        try:
+            c = mir_eval.chord.encode('C' + (len(q) > 0)*':' + q, reduce_extended_chords=True)
+        except:
```

This isn't the correct way to implement tests in pytest.

In general, a bare except (ie not catching a specific, expected exception type) is bad practice, since it lets any kind of error pass through.  Additionally, this test function will succeed no matter what; the log will contain your print outputs, but pytest will consider it a success.

A better way to do this would be to 1) parametrize the test by `quals.keys()` (so each encode runs independently of the others), and 2) not try to catch exceptions.  That way, anything that does not encode properly will raise an exception and we can see exactly what the failure was.  Something like:

```python
@pytest.mark.parametrize('quality', list(mir_eval.chord.QUALITIES.keys()))
def test_chord_qualities(quality):
    if len(quality) > 0:
        mir_eval.chord.encode(f"C:{quality}"), reduce_extended_chords=True)
    else:
        mir_eval.chord.encode("C", reduce_extended_chords=True)
```

---

Thank you again for the help, @bmcfee , I really appreciate you taking the time to explain things and provide solutions.

I just copy-pasted your example for the test, I haven't used pytest before - I'll need to start using it as it seems...

I don't know how it works, but if it is possible or "not improper" in some way, anyone is welcome to "take the pull request over" and complete it in a proper way (I am not concerned in getting any credit for the pull request etc, I'm just interested in the result :D ).

---

Great, thanks!

The tests are passing, but codecov is still giving us trouble.  I'll fix that in a separate PR.

Otherwise, the only lingering issue is that the linting action is failing:

https://github.com/mir-evaluation/mir_eval/actions/runs/12933203903/job/36071425462?pr=397

I checked this and it's an easy enough fix that I'll just patch it in here.  (Problem is using single quotes instead of double.)